scripts/flattener.py

```diff
@@ -607,24 +608,22 @@ def set_ensembl(redundant, feature_keys):
 	cxg_adata_raw.var.drop(columns=[i for i in raw_cols if i!='gene_ids'], inplace=True)
 	if feature_keys == ['gene symbol']:
 		if 'gene_ids' in cxg_adata_raw.var.columns.to_list():
-			# Check for gene symbols that are redudant and have suffix
-			# WILL NEED TO MAKE SURE SPLITTING ON '.' IS STILL APPROPRIATE FOR FUTURE DATASETS
+			# Drop unmapped genes from normalized matrix
 			norm_index = set(cxg_adata.var.index.to_list())
 			raw_index = set(cxg_adata_raw.var.index.to_list())
-			drop_redundant_with_suffix = list(norm_index.difference(raw_index))
-			for unmapped in drop_redundant_with_suffix:
-				unmapped_split = unmapped.split(".")
-				if unmapped_split[0] not in redundant:
-					logging.info('ERROR:\t{}\tnot redundant but unmapped'.format(unmapped))
-			logging.info('drop_redundant_with_suffix\t{}\t{}'.format(len(drop_redundant_with_suffix), drop_redundant_with_suffix))
-
-			cxg_adata = cxg_adata[:, [i for i in cxg_adata.var.index.to_list() if i not in drop_redundant_with_suffix]]
-			cxg_adata_raw.var_names_make_unique()
+			drop_unmapped = list(norm_index.difference(raw_index))
+			cxg_adata = cxg_adata[:, [i for i in cxg_adata.var.index.to_list() if i not in drop_unmapped]]
+			logging.info('drop_unmapped\t{}'.format(drop_unmapped))
+
 			cxg_adata.var = pd.merge(cxg_adata.var, cxg_adata_raw.var, left_index=True, right_index=True, how='left', copy = True)
 			cxg_adata.var = cxg_adata.var.set_index('gene_ids', drop=True)
 			cxg_adata_raw.var  = cxg_adata_raw.var.set_index('gene_ids', drop=True)
 			cxg_adata.var.index.name = None
 			cxg_adata_raw.var.index.name = None
+
+			# Drop redundant by Ensembl ID
+			cxg_adata = cxg_adata[:, [i for i in cxg_adata.var.index.to_list() if i not in redundant]]
```

Comment:
Should we log this similar to drop_unmapped?

Reply:
Sounds good.